File: 프로그래머스/unrated/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = new int[2];
        for(String str : keyinput){
            if(str.equals("left")) answer[0] -= answer[0] > -(board[0]/2) ? 1 : 0;
            else if(str.equals("right")) answer[0] += answer[0] < (board[0]/2) ? 1 : 0;
            else if(str.equals("up")) answer[1] += answer[1] < (board[1]/2) ? 1 : 0;
            else if(str.equals("down")) answer[1] -= answer[1] > -(board[1]/2) ? 1 : 0;
        }
        return answer;
    }
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    var answer = [0, 0];
    for(let i = 0; i<keyinput.length; i++){
        if(keyinput[i] === 'left') answer[0] += answer[0] > -((board[0]-1)/2) ? -1 : 0;
        else if(keyinput[i] === 'right') answer[0] += answer[0] < ((board[0]-1)/2) ? 1 : 0;
        else if(keyinput[i] === 'up') answer[1] += answer[1] < ((board[1]-1)/2) ? 1 : 0;
        else if(keyinput[i] === 'down') answer[1] += answer[1] > -((board[1]-1)/2) ? -1 : 0;
    }
    return answer;
}